src/core/services/compiler/compiler.js

```diff
@@ -6,7 +6,22 @@
  */
 angular
   .module('material.core')
-  .service('$mdCompiler', MdCompilerService);
+  .service('$mdCompiler', MdCompilerService)
+  .provider('$$mdPreAssignBindings', PreAssignBindingsProvider);
+
+/**
+ * Provider that is used to report the preAssignBindingsEnabled state to the $mdCompiler service.
+ */
+function PreAssignBindingsProvider($compileProvider) {
+  // To avoid that the preAssignBindings state causes issues when upgrading from AngularJS 1.5 to
+  // AngularJS 1.6, AngularJS Material will only respect the preAssignBindingsEnabled state in 1.6
+  var respectPreAssignState = angular.version.major === 1 && angular.version.minor === 6;
```

Comment:
BTW, just in case we'd have to revert the removal and defer it to 1.8, it would be safer to check for `angular.version.minor > 5` instead of `angular.version.minor === 6`